docs/Installation-Guide.rst

```diff
@@ -1,6 +1,41 @@
 Installation Guide
 ==================
 
+Versioning
+~~~~~~~~~~
+
+LightGBM releases use a 3-part version number, with this format:
+
+.. code::
+
+   {major}.{minor}.{patch}
+
+This version follows a scheme called Intended Effort Versioning ("Effver" for short).
+Changes to a component of the version indicate how much effort it will likely take to update
+code using a previous version.
+
+* ``major`` = updating will require significant effort
+* ``minor`` = some effort
+* ``patch`` = no or very little effort
+
+This means that **new minor versions can contain breaking changes**, but these are typically
+small or limited to less-frequently-used parts of the project.
+
+When built from source on an unreleased commit, this version takes the following form:
+
+.. code::
+
+   {major}.{minor}.{patch}.99
+
+That ``.99`` is added to ensure that a version built from an unreleased commit is considered "newer"
```

Comment:
Maybe move paragraph about nightly builds here?
https://github.com/microsoft/LightGBM/blob/f91dcfee8baf5fe329bfddadf38577f6443fdf18/docs/Installation-Guide.rst?plain=1#L44-L46

Reply:
Sure, good idea! Moved that in https://github.com/microsoft/LightGBM/pull/6905/commits/0338d2cb9d2f51925bc59fa632a10a0a2dcc021e

If you wanted a slightly different placement, it's ok with me if you push directly to this branch.

Reply:
Perfect, thanks!